experiment/build/generate_cloudbuild.py

```diff
@@ -13,64 +13,127 @@
-    cloud_build_spec['images'] = []
-
-    for name, image in buildable_images.items():
-        step = {}
-        step['id'] = name
-        step['name'] = 'gcr.io/cloud-builders/docker'
-        step['args'] = []
-        step['args'] += ['--tag', image['tag']]
-        step['args'] += ['--cache-from', docker_registry + image['tag']]
-        step['args'] += ['--build-arg', 'BUILDKIT_INLINE_CACHE=1']
-        if 'build_arg' in image:
-            for build_arg in image['build_arg']:
-                step['args'] += ['--build-arg', build_arg]
-        if 'dockerfile' in image:
-            step['args'] += ['--file', image['dockerfile']]
-        step['args'] += [image['context']]
-        if 'depends_on' in image:
-            step['wait_for'] = []
-            for dep in image['depends_on']:
-                step['wait_for'] += [dep]
-        cloud_build_spec['images'].append(name)
+import os
+import posixpath
+
+from common import yaml_utils
+from common import experiment_utils
+from common import experiment_path as exp_path
+from common.utils import ROOT_DIR
+from experiment.build import build_utils
+
+DOCKER_IMAGE = 'docker:19.03.12'
+PROJECT_DOCKER_REGISTRY = 'gcr.io/fuzzbench'
+
+
+def get_experiment_tag_for_image(image_specs, tag_by_experiment=True):
+    """Returns the registry with the experiment tag for given image."""
+    tag = posixpath.join(experiment_utils.get_base_docker_tag(),
+                         image_specs['tag'])
+    if tag_by_experiment:
+        tag += ':' + experiment_utils.get_experiment_name()
+    return tag
+
+
+def coverage_steps(benchmark):
+    """Returns GCB run steps for coverage builds."""
+    coverage_binaries_dir = exp_path.filestore(
+        build_utils.get_coverage_binaries_dir())
+    steps = [{
+        'name':
+            DOCKER_IMAGE,
+        'args': [
+            'run', '-v', '/workspace/out:/host-out',
+            posixpath.join(experiment_utils.get_base_docker_tag(), 'builders',
+                           'coverage', benchmark) + ':' +
+            experiment_utils.get_experiment_name(), '/bin/bash', '-c',
+            'cd /out; tar -czvf /host-out/coverage-build-' + benchmark +
+            '.tar.gz * /src /work'
+        ]
+    }]
+    step = {'name': 'gcr.io/cloud-builders/gsutil'}
+    step['args'] = [
+        '-m', 'cp', '/workspace/out/coverage-build-' + benchmark + '.tar.gz',
+        coverage_binaries_dir + '/'
+    ]
+    steps.append(step)
+    return steps
+
+
+def create_cloud_build_spec(image_templates,
+                            benchmark='',
+                            build_base_images=False):
+    """Generates Cloud Build specification.
+
+    Args:
+      image_templates: Image types and their properties.
+      benchmark: Name of benchmark (required for coverage builds only).
+      build_base_images: True if building only base images.
+
+    Returns:
+      GCB build steps.
+    """
+    cloud_build_spec = {'steps': [], 'images': []}
+
+    if build_base_images:
```

Comment:
This is added condition for explicit pull.

Reply:
Was this only an issue for base images? If yes, consider extending the comment explaining why we need this workaround only for the `build_base_images` case.